### PR TITLE
p4-fusion: remove duplicated command building

### PR DESCRIPTION
**Pull request by mrnugget**
Ran into this with @burmudar while looking at the code. It's easy for this to drift apart.

## Test plan

- Existing tests

### CODE REVIEW

**Review by burmudar — APPROVED** [review]

File: cmd/gitserver/server/vcs_syncer_perforce.go [hunk 1]
```diff
@@ -88,6 +72,26 @@ func (s *PerforceDepotSyncer) CloneCommand(ctx context.Context, remoteURL *vcs.U
...
+func (s *PerforceDepotSyncer) buildP4FusionCmd(ctx context.Context, depot, username, src, host string) *exec.Cmd {
+	// Example: p4-fusion --path //depot/... --user $P4USER --src clones/ --networkThreads 64 --printBatch 10 --port $P4PORT --lookAhead 2000 --retries 10 --refresh 100
+	return exec.CommandContext(ctx, "p4-fusion",
+		"--path", depot+"...",
+		"--client", s.FusionConfig.Client,
+		"--user", username,
+		"--src", src,
+		"--networkThreads", strconv.Itoa(s.FusionConfig.NetworkThreads),
+		"--printBatch", strconv.Itoa(s.FusionConfig.PrintBatch),
+		"--port", host,
```
Comment (sashaostrikov) [inline cmd/gitserver/server/vcs_syncer_perforce.go:84]: I understand that it is the way it was before, but it is a bit confusing to add `host` as a `--port` flag 😀

**Review by sashaostrikov — APPROVED** [review]
LGTM, added a comment about host-port thing

File: cmd/gitserver/server/vcs_syncer_perforce.go (hunk 1 above)
Comment (ryanslade) [reply cmd/gitserver/server/vcs_syncer_perforce.go:84]: This is a Perforce thing.

https://www.perforce.com/manuals/cmdref/Content/CmdRef/P4PORT.html

> For Helix Server applications, the protocol, host and port number of the Perforce service with which to communicate. The most commonly-used communications protocols are tcp (plaintext over TCP/IP) or ssl (SSL over TCP/IP).

**Review by ryanslade — APPROVED** [review]

File: cmd/gitserver/server/vcs_syncer_perforce.go (hunk 1 above)
Comment (mrnugget) [reply cmd/gitserver/server/vcs_syncer_perforce.go]: Agree with @sashaostrikov though. In this context it _is_ the p4 port. It was the host in the URL from which it came, but it's clearer to call it port here. I pushed some renames to this branch.

**Pull request merged**